source/fundamentals/builders/aggregates.txt

```diff
@@ -787,6 +787,47 @@ and ``temperature`` fields:
    :language: java
    :dedent:
 
+Densify
+-------
+
+Use the ``densify()`` method to create a
+`$densify <https://www.mongodb.com/docs/rapid/reference/operator/aggregation/densify/>`__
+pipeline stage that generates a sequence of documents to span a specified
+interval
+
+.. tip::
+
+   You can only use the ``$densify()`` aggregation stage when running
+   MongoDB v5.1 or later.
+
+The following example creates a pipeline stage that adds documents every
+15 minutes for which ``ts`` values do not already exist. The partition option
+specifies that the stage should group the documents using the ``position``
+field.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggregateDensify.java
+   :start-after: // begin densify aggregate
+   :end-before: // end densify aggregate
+   :language: java
+   :dedent:
+
+The following output highlights the documents added by the aggregate stage
+which contain``ts`` values every 15 minutes between the existing documents:
+
+.. code-block:: java
+   :emphasize-lines: 2-4
+   :copyable: false
+
+   Document{{ _id=5553a..., position=Document{{type=Point, coordinates=[-47.9, 47.6]}}, ts=Mon Mar 05 08:00:00 EST 1984, ... }}
+   Document{{ position=Document{{coordinates=[-47.9, 47.6]}}, ts=Mon Mar 05 08:15:00 EST 1984 }}
+   Document{{ position=Document{{coordinates=[-47.9, 47.6]}}, ts=Mon Mar 05 08:30:00 EST 1984 }}
+   Document{{ position=Document{{coordinates=[-47.9, 47.6]}}, ts=Mon Mar 05 08:45:00 EST 1984 }}
+   Document{{ _id=5553b..., position=Document{{type=Point, coordinates=[-47.9, 47.6]}}, ts=Mon Mar 05 09:00:00 EST 1984, ... }}
```

Comment:
I definitely considered that. 

I thought it might not make much sense given how the rest of the Builder entries are structured on the page (no sample data, no output) as they defer to the Manual docs for more explanation. That said, it looks like there's a mix throughout the builders pages (Aggregates and Index Builders do not show input/output, but the rest of the Builders pages do).

I came up with an in between solution as a compromise. That said, maybe starting the pattern of showing input data and output data just in case a reader expects it. Aggregate sample data is slightly more complex because of the variety of pipeline operators that require different document structures and potentially large sample data sets.

For this PR, aligned with your feedback, I will add the sample set to just this example since it's very compact, but maybe we need to reconsider what we show on Builders pages. Will add a team sync discussion.

Reply:
This looks good to me.